platform/android/export/export.cpp

```diff
@@ -1463,6 +1466,67 @@ class EditorExportPlatformAndroid : public EditorExportPlatform {
 		}
 	}
 
+	void _copy_icons_to_gradle_project(const Ref<EditorExportPreset> &p_preset) {
+		String project_icon_path = ProjectSettings::get_singleton()->get("application/config/icon");
+		// Prepare images to be resized for the icons. If some image ends up being uninitialized,
+		// the default image from the export template will be used.
+		Ref<Image> launcher_icon_image;
+		Ref<Image> launcher_adaptive_icon_foreground_image;
+		Ref<Image> launcher_adaptive_icon_background_image;
+
+		launcher_icon_image.instance();
+		launcher_adaptive_icon_foreground_image.instance();
+		launcher_adaptive_icon_background_image.instance();
+
+		// Regular icon: user selection -> project icon -> default.
+		String path = static_cast<String>(p_preset->get(launcher_icon_option)).strip_edges();
+		if (path.empty() || ImageLoader::load_image(path, launcher_icon_image) != OK) {
+			ImageLoader::load_image(project_icon_path, launcher_icon_image);
+		}
+
+		// Adaptive foreground: user selection -> regular icon (user selection -> project icon -> default).
+		path = static_cast<String>(p_preset->get(launcher_adaptive_icon_foreground_option)).strip_edges();
+		if (path.empty() || ImageLoader::load_image(path, launcher_adaptive_icon_foreground_image) != OK) {
+			launcher_adaptive_icon_foreground_image = launcher_icon_image;
+		}
+
+		// Adaptive background: user selection -> default.
+		path = static_cast<String>(p_preset->get(launcher_adaptive_icon_background_option)).strip_edges();
+		if (!path.empty()) {
+			ImageLoader::load_image(path, launcher_adaptive_icon_background_image);
+		}
+
+		for (int i = 0; i < icon_densities_count; ++i) {
+			if (launcher_icon_image.is_valid() && !launcher_icon_image->empty()) {
+				Vector<uint8_t> data;
+				_process_launcher_icons("", data, launcher_icon_image, launcher_icons[i], true);
+				String img_path = launcher_icons[i].export_path;
+				img_path = img_path.insert(0, "res://android/build/");
+				store_file_at_path(img_path, data);
+			}
```

Comment:
You can wrap this into a small utility function since the same structure is repeated below for `launcher_adaptive_icon_foreground_image` and `launcher_adaptive_icon_background_image`.